src/coreclr/debug/daccess/request_svr.cpp

```diff
@@ -459,4 +459,26 @@ HRESULT DacHeapWalker::InitHeapDataSvr(HeapData *&pHeaps, size_t &pCount)
     return S_OK;
 }
 
+void DacFreeRegionEnumerator::AddServerRegions()
+{
+    // Cap the number of free regions we will walk at a sensible number.  This is to protect against
+    // memory corruption, un-initialized data, or just a bug.
+    int count_free_region_kinds = g_gcDacGlobals->count_free_region_kinds;
+    count_free_region_kinds = min(count_free_region_kinds, 16);
+
+    for (int i = 0; i < GCHeapCount(); i++)
+    {
+        TADDR heapAddress = (TADDR)HeapTableIndex(g_gcDacGlobals->g_heaps, i);
```

Comment:
At this point, we know which heap the free region belongs to, shall we keep track of that in `ExtraData` together with the free region type? This will allow us to diagnose free list imbalance in case that happens.

Reply:
We use that for the region kind.  I should probably add a simple "heap #" field to account for this.